l10n_ch_lsv_dd/wizard/dd_export_wizard.py

```diff
@@ -0,0 +1,497 @@
+        vals['reserve_11'] = self._complete_line('', 3)
+        vals['reserve_12'] = self._complete_line('', 1)
+        vals['orderer_info'] = self._complete_line('', 140)  # Not implemented
+        vals['reserve_13'] = self._complete_line('', 14)
+
+        debit_record = control_range + ''.join(vals.values())
+
+        if len(debit_record) == 700:  # Standard debit record size
+            return debit_record
+        else:
+            raise exceptions.Warning(
+                _('Generated debit_record with size %d is not valid '
+                  '(len should be 700)') % len(debit_record)
+            )
+
+    @api.model
+    def _generate_total_record(self, properties, total_amount):
+        ''' Generate total line according to total amount and properties
+            (Transaction type 97)
+        '''
+        control_range = self._gen_control_range('97', properties)
+
+        vals = collections.OrderedDict()
+        vals['currency'] = properties.get('currency', 'CHF')
+        vals['nb_transactions'] = str(
+            properties.get('trans_ser_no') -
+            1).zfill(6)
+        vals['total_amount'] = self._format_number(total_amount, 13)
+        vals['reserve_1'] = self._complete_line('', 628)
+
+        total_record = control_range + ''.join(vals.values())
+        if len(total_record) == 700:
+            return total_record
+        else:
+            raise exceptions.Warning(
+                _('Generated total line is not valid (%d instead of 700)') %
+                len(total_record)
+            )
+
+    @api.model
+    def _create_dd_export(self, p_o_ids, total_amount,
+                          properties, file_content):
+        ''' Create banking.export.ch.dd object '''
+        banking_export_ch_dd_obj = self.env['banking.export.ch.dd']
+        vals = {
+            'payment_order_ids': [(6, 0, [p_o_id for p_o_id in p_o_ids])],
+            'total_amount': total_amount,
+            'nb_transactions': properties.get('nb_transactions'),
+            'file': base64.encodestring(file_content),
+            'type': 'Postfinance Direct Debit',
+        }
+        export_id = banking_export_ch_dd_obj.create(vals)
+        return export_id
+
+    @api.one
+    def confirm_export(self):
+        ''' Save the exported DD file: mark all payments in the file
+            as 'sent'. Write 'last debit date' on mandate.
+        '''
+        self.banking_export_ch_dd_id.write({'state': 'sent'})
+        wf_service = netsvc.LocalService('workflow')
+        today_str = datetime.today().strftime(DF)
+        for order in self.banking_export_ch_dd_id.payment_order_ids:
+            wf_service.trg_validate(self.env.uid, 'payment.order', order.id,
+                                    'done', self.env.cr)
+            mandate_ids = list(set([line.
+                                    mandate_id.id for line in order.line_ids]))
+            mandates = self.env['account.banking.mandate'].browse(mandate_ids)

```

Comment:
You might prefer to use `mapped()`

    mandates = order.mapped('line_ids.mandate_id').ids

Then do a search for uniquness:

    self.env['account.banking.mandate'].search(['id', 'in', mandate.ids])

An other option to create directly a record set is to do

    mandates = self.env['account.banking.mandate'].browse()
    for something:
        mandates |= a_mandate